front/src/components/ProductCatalogue/ProductRating.js

```diff
@@ -1,41 +1,91 @@
 import React, { useEffect, useState } from "react";
-import { Box, Rating } from "@material-ui/core";
+import { Box } from "@material-ui/core";
+import axios from "axios";
+import Rating from "@material-ui/lab/Rating";
+import { useSelector } from "react-redux";
 
 function ProductRating(props) {
-  // const { user } = useSelector(state => state.account);
-  const [rating, setRating] = useState(0);
+  const { user } = useSelector((state) => state.account);
+  const [avgRating, setAvgRating] = useState(2);
+  const [count, setCount] = useState(1);
+  const [isRated, setIsRated] = useState(false); //flag for user input
+  const [isReadOnly, setIsReadOnly] = useState(false);
+
+  const handleIsReadOnly = (value) => {
```

Comment:
It seems that this sends the requests to the back-end immediately after the user signs in. Not sure if this is the right approach

Reply:
line 26. On each render you will call const handleIsReadOnly. 

can be refactored, try this. 

 const [isReadOnly, setIsReadOnly] = useState(user ? false : true);